### PR TITLE
Fix viewId not being picked up by ComponentEventParams

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?
A change made in #22301 removed the `viewId` from the `ComponentEventParams`, this caused an issue where we could no longer track `SIGN_IN` and `CREATE_ACCOUNT` events in identity.

This reintroduces `viewId`, and makes it a required value in the `ComponentEventParams` type too.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/sign-in-gate/types.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export type ComponentEventParams = {
     componentId?: string,
     abTestName: string,
     abTestVariant: string,
-    viewId?: string,
```
Comment (vlbee) [inline static/src/javascripts/projects/common/modules/identity/sign-in-gate/types.js:13]: Why was this made optional in the original PR? Hard to parse.

**Review by vlbee — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/identity/sign-in-gate/index.js [hunk 2]
```diff
@@ -60,12 +60,23 @@ const show: () => Promise<boolean> = () =>
             `${config.get('page.host')}/${config.get('page.pageId')}`
         );
 
+        // get the view id to attach to component event params
+        let viewId = '';
+        if (
+            window.guardian &&
```
Comment (vlbee) [inline static/src/javascripts/projects/common/modules/identity/sign-in-gate/index.js:66]: it would be so nice if frontend supported new optional chaining syntax...

File: static/src/javascripts/projects/common/modules/identity/sign-in-gate/types.js (hunk 1 above)
Comment (coldlink) [reply static/src/javascripts/projects/common/modules/identity/sign-in-gate/types.js:13]: i have a feeling that this was a mistake on my part when initially setting up the sign in gate, since this parameter is required in identity api to fire the component event

File: static/src/javascripts/projects/common/modules/identity/sign-in-gate/index.js (hunk 2 above)
Comment (coldlink) [reply static/src/javascripts/projects/common/modules/identity/sign-in-gate/index.js:66]: ikr would be so much nicer! 🙄
